src/lib/iframe-tools/iframe-manager.js

```diff
@@ -0,0 +1,45 @@
+import createIframe from './create-iframe'
+
+export class IframeManager {
+  constructor(config) {
+    this.url = new URL(config.src)
+    this.iframe = createIframe(config.src)
+    this.messageListeners = config.messageListeners || [];
+ 
+    this.iframe.addEventListener('load', this._onIframeLoad)
+    window.addEventListener('message', this._onMessage)
+  }
+
+  _onIframeLoad = (e) => {
+    if (this.onIframeLoad) {
+      this.onIframeLoad(e)
+    }
+  }
+
+  _onMessage = (e) => {
+    if (e.source.window !== this.iframe.contentWindow) {
+      return
+    }
+
+    if (e.origin !== this.url.origin) {
+      return
+    }
+
+    for (let i = 0; i < this.messageListeners.length; i++) {
```

Comment:
whats the reason of not using a `forEach` here?

Reply:
No reason at all. I'll update it.